pkg/machine/qemu/machine.go

```diff
@@ -974,7 +1014,31 @@ func (v *MachineVM) Stop(_ string, _ machine.StopOptions) error {
 	if err := v.update(); err != nil {
 		return err
 	}
-	return v.stopLocked()
+	
+	stopErr := v.stopLocked()
+
+	// Make sure that the associated QEMU process gets killed in case it's
+	// still running (#16054).
+	qemuRunning, qemuPid, err := v.qemuProcessRunning()
+	if err != nil {
+		if stopErr == nil {
+			return err
+		}
+		return fmt.Errorf("%w: %w", stopErr, err)
+	}
+
+	if !qemuRunning {
+		return stopErr
+	}
+
+	if err := unix.Kill(qemuPid, 9); err != nil {
```

Comment:
Please us a constant for the signal number.
While 9 seems to be used everywhere for sigkill we should never assume signal numbers are the same across linux/macos/bsd, etc...